bemserver_ui/static/scripts/modules/views/timeseries/dataExplore.js

```diff
@@ -65,11 +65,77 @@ export class TimeseriesDataExploreView {
         this.#endDatetimePickerElmt.dateMin = this.#startDatetimePickerElmt.date;
     }
 
+    #addTsParamInputs(tsData) {
+        let tsParam = document.createElement("div");
+        tsParam.id = "tsParam-" + tsData.itemId;
+        
+        tsParam.innerHTML =
```

Comment:
This HTML content is passed as text. It works fine, and later when we want to get input values we use `document.getElementById(...)` instructions.
I usually create dynamically and manipulate HTML elements with JS code (using `document.createElement` instruction and so on...).
I don't really know if this way is better or worse, but the advantage to create things through code, and keeping references in private class attributes, allows to access them (read/update...) without searching those references with `document.getElemetById(...)` each time. Like it is done here below, from actual line 233, in the `for` loop.
Maybe something like a dict could contain each reference of parameter input instances in a timeseries id key. And we could access quickly them later.
Code is a little longer to write, but it might be worth it.

Example:
```
// New view class attribute.
#tsParamInputElmts = {};
```
Code inside `#addTsParamInputs(tsData)` method:
```
// keep line 69 (createElement `tsParam` div)
// and then create all other timeseries input elements (id and name of those HTML elements are not required anymore, as we will directly keep instances reference):
let tsYAxis = document.createElemen("select");
tsYAxis.classList.add("form-select");
for (let [yAxisOptVal, yAxisOptTxt] of Object.entries({"left": "Left", "right": "Right"}) {
    let yAxisOpt = document.createElement("option");
    yAxisOpt.value = yAxisOptVal;
    yAxisOpt.innerText = yAxisOptTxt;
    yAxisOpt.selected = yAxisOptVal == "left";
    tsYAxis.appendChild(yAxisOpt);
}
tsYAxisInputGroup.appendChild(tsYAxis);
let tsType = document.createElement("select");
// and so on...

// then update the `#tsParamInputElmts` class attribute.
this.#tsParamInputElmts[tsData.itemId] = {
    "containerElmt": tsParam,
    "yAxisInputElmt": tsYAxis,
    "typeInputElmt": tsType,
    ...
}
```

If a timeseries is unselected, then just remove container and the corresponding entry in the dict:
```
this.#tsParamInputElmts[tsId]["containerElmt"].remove();
delete this.#tsParamInputElmts[tsId];
```

Finally later in the code, inside `#refreshChart` function, around line 233, access to timeseries parameter input values, in `for` loop using this dict:
```
options.series[tsName] = {
    position: this.#tsParamInputElmts[tsId]["yAxisInputElmt"].value,
    ....,
}
```